src/packageurl/contrib/django_models.py

```diff
@@ -114,16 +108,20 @@ def package_url(self):
     def set_package_url(self, package_url):
         """
         Set each field values to the values of the provided `package_url` string
-        or PackageURL object. Existing values are overwritten including setting
-        values to None for provided empty values.
+        or PackageURL object.
+        Existing values are always overwritten, forcing the new value or an
+        empty string on all the `package_url` fields since we do not want to
+        keep any previous values.
         """
         if not isinstance(package_url, PackageURL):
             package_url = PackageURL.from_string(package_url)
 
-        for field_name, value in package_url.to_dict(encode=True).items():
+        package_url_dict = package_url.to_dict(encode=True, empty='')
+        for field_name, value in package_url_dict.items():
             model_field = self._meta.get_field(field_name)
 
             if value and len(value) > model_field.max_length:
-                raise ValidationError(_('Value too long for field "{}".'.format(field_name)))
+                message = _('Value too long for field "{}".'.format(field_name))
+                raise ValidationError(message)
 
-            setattr(self, field_name, value or None)
+            setattr(self, field_name, value)
```

Comment:
Have you considered `setattr(self, field_name, value or '')`?

Reply:
Yes I did. It was my initial approach, but the empty value is now controlled in the above `to_dict` which makes it unneeded in the `setattr`.